bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1173,20 +1178,95 @@ public void smartHomeCommand(String entityId, String action, @Nullable String pr
         JsonObject parameters = new JsonObject();
         parameters.addProperty("action", action);
         if (property != null) {
-            if (value instanceof QuantityType<?>) {
-                parameters.addProperty(property + ".value", ((QuantityType<?>) value).floatValue());
-                parameters.addProperty(property + ".scale",
-                        ((QuantityType<?>) value).getUnit().equals(SIUnits.CELSIUS) ? "celsius" : "fahrenheit");
-            } else if (value instanceof Boolean) {
-                parameters.addProperty(property, (boolean) value);
-            } else if (value instanceof String) {
-                parameters.addProperty(property, (String) value);
-            } else if (value instanceof Number) {
-                parameters.addProperty(property, (Number) value);
-            } else if (value instanceof Character) {
-                parameters.addProperty(property, (Character) value);
-            } else if (value instanceof JsonElement) {
-                parameters.add(property, (JsonElement) value);
+            if ("setThermostatMode".equals(action)) {
+                if (value instanceof StringType) {
+                    parameters.addProperty(property + ".value", value.toString());
+                }
+            } else if ("setTargetTemperature".equals(action)) {
+                if ("targetTemperature".equals(property)) {
+                    if (value instanceof QuantityType<?>) {
+                        parameters.addProperty(property + ".value", ((QuantityType<?>) value).floatValue());
+                        parameters.addProperty(property + ".scale",
+                                ((QuantityType<?>) value).getUnit().equals(SIUnits.CELSIUS) ? "celsius" : "fahrenheit");
+                    }
+                } else {
+                    Map<String, JsonArray> devices = null;
+                    try {
+                        devices = getSmartHomeDeviceStatesJson(new HashSet<>(getSmarthomeDeviceList()));
+                    } catch (URISyntaxException e) {
+                        logger.error("{}", e.toString());
```

Comment:
Will this stop the binding from working correctly after this, i.e. considered a major error? Otherwise log level should be reduced. For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
I've changed this to debug.